From 9a3d32d5180626c70aab7733d7eda9c2d307cb98 Mon Sep 17 00:00:00 2001
From: Peng Liang <liangpeng10@huawei.com>
Date: Mon, 8 Feb 2021 17:16:34 +0800
Subject: [PATCH] kvm: arm64: add KVM_CAP_ARM_CPU_FEATURE extension
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 9a3d32d5180626c70aab7733d7eda9c2d307cb98
Modified-by-SEL: Yes, refreshed due to different context


hulk inclusion
category: feature
bugzilla: 48052
CVE: NA

------------------------------

Add KVM_CAP_ARM_CPU_FEATURE extension for userpace to check whether KVM
supports to set CPU features in AArch64.

Signed-off-by: zhanghailiang <zhang.zhanghailiang@huawei.com>
Signed-off-by: Peng Liang <liangpeng10@huawei.com>
Reviewed-by: Zhanghailiang <zhang.zhanghailiang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kvm/arm.c     |    1 +
 include/uapi/linux/kvm.h |    2 ++
 2 files changed, 3 insertions(+)

--- a/arch/arm64/kvm/arm.c
+++ b/arch/arm64/kvm/arm.c
@@ -198,6 +198,7 @@ int kvm_vm_ioctl_check_extension(struct
 	case KVM_CAP_ARM_IRQ_LINE_LAYOUT_2:
 	case KVM_CAP_ARM_NISV_TO_USER:
 	case KVM_CAP_ARM_INJECT_EXT_DABT:
+	case KVM_CAP_ARM_CPU_FEATURE:
 		r = 1;
 		break;
 	case KVM_CAP_ARM_SET_DEVICE_ADDR:
--- a/include/uapi/linux/kvm.h
+++ b/include/uapi/linux/kvm.h
@@ -1063,6 +1063,8 @@ struct kvm_ppc_resize_hpt {
 #define KVM_CAP_ENFORCE_PV_FEATURE_CPUID 190
 #define KVM_CAP_SGX_ATTRIBUTE 196
 
+#define KVM_CAP_ARM_CPU_FEATURE 555
+
 #ifdef KVM_CAP_IRQ_ROUTING
 
 struct kvm_irq_routing_irqchip {
